### PR TITLE
Upgrade node to 18.12.0

### PR DESCRIPTION
**Pull request by andrewvc**
Update the project to use node `18.12.0`, also updates the github actions workflows to match the new ones in the synthetics-demo repo.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@test/node') _
```
Comment (v1v) [inline .ci/Jenkinsfile:3]: This is just for testing purposes, so the nvm/npm/nodejs installation that happens on the fly will report back what some details about what it has been installed. See https://github.com/elastic/apm-pipeline-library/pull/1928/files#diff-afc97124fbb38137b00b422a1f1b72d6a458857071329f80456684ee4891d280

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
 pipeline {
-  agent { label 'ubuntu-18 && immutable' }
+  agent { label 'ubuntu-2204 && immutable' }
```
Comment (v1v) [inline .ci/Jenkinsfile:6]:
```suggestion
  agent { label 'ubuntu-22 && immutable' }
```

File: .ci/Jenkinsfile (hunk 1 above)
Comment (v1v) [reply .ci/Jenkinsfile:3]:
```suggestion
@Library('apm@current') _
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -3,7 +3,7 @@
 @Library('apm@current') _
 
 pipeline {
-  agent { label 'ubuntu-18 && immutable' }
+  agent { label 'ubuntu-22 && immutable' }
```
Comment (v1v) [inline .ci/Jenkinsfile:6]:
```suggestion
  agent { label 'ubuntu-20 && immutable' }
```

**Review by vigneshshanmugam — APPROVED** [review]

File: __tests__/push/bundler.test.ts [hunk 4]
```diff
@@ -46,18 +46,27 @@ async function validateZip(content) {
...
   const files: Array<string> = [];
 
-  const entries = createReadStream(pathToZip).pipe(
-    unzipper.Parse({ forceStream: true })
-  );
-
-  let targetFileContent = null;
-  for await (const entry of entries) {
-    files.push(entry.path);
+  console.log(partialPath);
```
Comment (lucasfcosta) [inline __tests__/push/bundler.test.ts:49]: I think you might have forgotten this one.

File: __tests__/push/bundler.test.ts [hunk 5]
```diff
@@ -46,18 +46,27 @@ async function validateZip(content) {
...
+      .on('close', () => {
+        r(null);
+      });
+  });
 
-    if (entry.path === partialPath) {
-      entry.on('data', d => (targetFileContent += d));
-    }
-  }
+  console.log('TARGET CONTENTS', targetFileContent);
```
Comment (lucasfcosta) [inline __tests__/push/bundler.test.ts:69]: [+1] I think you might have forgotten this one.

File: __tests__/push/bundler.test.ts [hunk 6]
```diff
@@ -46,18 +46,27 @@ async function validateZip(content) {
...
+        files.push(fileName);
+        if (fileName === partialPath) {
+          entry.on('data', d => (targetFileContent += d));
+        } else {
+          entry.autodrain();
+        }
+      })
+      .on('close', () => {
+        r(null);
+      });
```
Comment (lucasfcosta) [inline __tests__/push/bundler.test.ts:66]: We can also just resolve here:

```suggestion
      .on('close', r)
```

**Review by lucasfcosta — APPROVED** [review]
LGTM. Although I've left comments on the `console.log`s I think I can leave the approval for now and we can merge once those two are removed.

File: __tests__/plugins/network.test.ts [hunk 7]
```diff
@@ -265,9 +265,8 @@ describe('network', () => {
       waitUntil: 'networkidle',
     });
     await driver.page.reload({ waitUntil: 'networkidle' });
-    await delay(delayTime);
+    await delay(delayTime + 1);
     const netinfo = await network.stop();
-    await Gatherer.stop();
```
Comment (vigneshshanmugam) [inline __tests__/plugins/network.test.ts:270]: dont we need this?

Comment (andrewvc) [reply __tests__/plugins/network.test.ts:270]: This causes a test failure with node 18, it's actually  double close, something closes it earlier

Comment (andrewvc) [reply __tests__/plugins/network.test.ts:270]: Actually, I take it back, this test is acting funky with and without this. Could you take a look when you have a second?

**Pull request merged**